pyccel/codegen/printing/pycode.py

```diff
@@ -83,10 +83,13 @@ def _print_FunctionDef(self, expr):
         imports = self._indent_codestring(imports)
         code = ('def {name}({args}):\n'
                 '\n{imports}\n{body}\n').format(name=name, args=args,imports=imports, body=body)
-
         decorators = expr.decorators
-
         if decorators:
+            if decorators['template']:
+                #Eliminate templates_dict to avoid conflicts in the printing
+                expr.decorators['template'] = expr.decorators['template']['template_list']
+            else:
+                expr.decorators.pop('template')
```

Comment:
Can you explain what the pop is for? Does it remove an empty dict? Is `template` always present in `decorators` or might this raise a `KeyError`?

Reply:
`template` is `None` if there are no templates